### PR TITLE
Chore: Making versioncheck url rely on config instead of being hardcoded

### PR DESCRIPTION
**Pull request by tolzhabayev**
**What is this feature?**
Makes versioncheck url respect the config settings instead of being hardcoded. This should help us have clear plugin catalog environment separation.

It should have no effect on anyone who does not have grafanacom url configured differently than the defaults.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-plugins-platform-team/issues/115

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/updatechecker/plugins.go [hunk 1]
```diff
@@ -113,12 +115,13 @@ func (s *PluginsService) instrumentedCheckForUpdates(ctx context.Context) {
...
 	ctxLogger := s.log.FromContext(ctx)
-	ctxLogger.Debug("Checking for updates")
+	ctxLogger.Debug("Preparing plugins eligible for version check")
 	localPlugins := s.pluginsEligibleForVersionCheck(ctx)
-	requestURL := "https://grafana.com/api/plugins/versioncheck?" + url.Values{
+	requestURL := s.grafanaComAPIURL + "/plugins/versioncheck?" + url.Values{
 		"slugIn":         []string{s.pluginIDsCSV(localPlugins)},
 		"grafanaVersion": []string{s.grafanaVersion},
 	}.Encode()
+	ctxLogger.Debug("Checking for plugin updates via " + requestURL)
```
Comment (wbrowne) [inline pkg/services/updatechecker/plugins.go:124]:
```suggestion
	ctxLogger.Debug("Checking for plugin updates", "url", requestURL)
```

**Review by wbrowne — APPROVED** [review]

**Review by andresmgot — APPROVED** [review]
FYI, if you want the ephemeral instance to work, you need to merge `main` into this branch.

**Pull request merged**